### PR TITLE
Web list doctors

### PR DESCRIPTION
**Pull request by basicem**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -125,3 +125,5 @@ dist
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+.vscode/
```
Comment (amarbasic) [inline .gitignore]: Add new line

File: jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "compilerOptions": {
+    "baseUrl": "src"
+    },
+    "include": ["src"]
+}
```
Comment (amarbasic) [inline jsconfig.json]: Add new line

File: src/api/api.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+export default axios.create({
```
Comment (amarbasic) [inline src/api/api.js]: create variable then export:

```
const API = ...

export default API;
```

File: src/components/Doctors/DoctorCard.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  margin: 1rem;
+  border: 1px #f4f0ec solid;
+  box-shadow: 5px 5px 5px #f4f0ec;
+`;
+
+const TopDetails = styled.div`
+  flex: 4;
+  padding-top: 10px;
+  padding-bottom: 10px;
+`;
```
Comment (amarbasic) [inline src/components/Doctors/DoctorCard.js:32]: Add new line

File: src/components/Doctors/Doctors.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from "react";
+
+import DoctorList from "components/Doctors/DoctorList";
+import { getDoctors } from "api/doctors";
+
+const Doctors = () => {
+  const [count, setCount] = useState(0);
+  const [nextPage, setNextPage] = useState(0);
+  const [rows, setRows] = useState(undefined);
```
Comment (amarbasic) [inline src/components/Doctors/Doctors.js]: `undefined` is the default, no need to specify it

**Review by amarbasic — COMMENTED** [review]
looks good.

missing loading

**Comment by basicem** [comment]
can you check loader

File: src/components/Doctors/DoctorList.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (!rows) {
+    return null;
+  }
+
+  if (rows.length === 0) {
+    return (
+      <StyledContainer>
+        <Segment inverted color="red" secondary>
+          Doctors not find!
```
Comment (amarbasic) [inline src/components/Doctors/DoctorList.js]: *not found

File: src/components/Doctors/Doctors.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  const [active, setActive] = useState(true);
+
+  useEffect(() => {
+    const fetch = async () => {
+      const response = await getDoctors();
+      setCount(response.count);
+      setNextPage(response.nextPage);
+      setRows(response.rows);
+      setError(response.error);
+      setActive(false);
```
Comment (amarbasic) [inline src/components/Doctors/Doctors.js]: use `loading` naming as it's more descriptive instead of `active`

File: src/components/MyLoader.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import styled from "styled-components";
+
+const StyledContainer = styled(Container)`
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: column;
+  margin: 1rem;
+`;
+
+const MyLoader = ({ isActive }) => (
```
Comment (amarbasic) [inline src/components/MyLoader.js]: Don't use `My`, `Custom`, etc. prefixes.

Just call it `Loader` or `Spinner`

If you have a collision with import you can:
```
 import { Container, Segment, Dimmer, Loader as SemanticLoader, Image } from "semantic-ui-react";
```

File: src/components/Doctors/Doctors.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+const Doctors = () => {
+  const [count, setCount] = useState(0);
+  const [nextPage, setNextPage] = useState(0);
+  const [rows, setRows] = useState(null);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetch = async () => {
+      const response = await getDoctors();
```
Comment (amarbasic) [inline src/components/Doctors/Doctors.js]: you are missing try-catch. What if the request fails? The user will not see an error message.

File: src/components/Doctors/Doctors.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      setCount(response.count);
+      setNextPage(response.nextPage);
+      setRows(response.rows);
+      setError(response.error);
+      setLoading(false);
+    };
+
+    fetch();
+  }, []);
+  return (
```
Comment (amarbasic) [inline src/components/Doctors/Doctors.js:31]: Add new line before

**Review by amarbasic — APPROVED** [review]

**Pull request merged**